pkg/apis/minio.min.io/v2/helper.go

```diff
@@ -761,6 +761,26 @@ func (t *Tenant) CreateUsers(madmClnt *madmin.AdminClient, userCredentialSecrets
 	return nil
 }
 
+// CheckBucketsExist checks if the given buckets exist in the MinIO server
+func (t *Tenant) CheckBucketsExist(minioClient *minio.Client, buckets ...Bucket) error {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*20)
+	defer cancel()
+	allBuckets, err := minioClient.ListBuckets(ctx)
+	if err != nil {
+		return err
+	}
+	allBucketsMap := make(map[string]bool)
+	for _, bucket := range allBuckets {
+		allBucketsMap[bucket.Name] = true
+	}
+	for _, bucket := range buckets {
+		if !allBucketsMap[bucket.Name] {
```

Comment:
I think we can just use `slices.Contains(allBuckets, bucket)` for this check. No need to create a map.

Reply:
Not really. 
```go
// Contains reports whether v is present in s.
func Contains[S ~[]E, E comparable](s S, v E) bool {
	return Index(s, v) >= 0
}
```
The args must have the same type. @shtripat

Reply:
If it's not possible, and since it's currently functioning, I'll leave it as it is. Alternatively, we can create a new pull request to make further improvements as needed.